### ISSUE TITLE
Server crashes on boot-up when bad token is included.

### ISSUE THREAD

**Issue by gabeodess** [exciting-io/slack-bot-server#24]
Using the advanced example, we loop through the bot tokens in our database and add them before the server starts.  However, if there is a bad token, the whole server crashes.  The server should instead log a warning for the bad token, and continue to boot.

## Code
```ruby
queue = SlackBotServer::RedisQueue.new

server = SlackBotServer::Server.new(queue: queue)

server.on_add do |token|
  VinnyBot.new(token: token)
end

User.find_each do |user|
  server.add_bot(user.bot_token) if user.bot_token.present?
end

server.start

```

## Backtrace
```
2016-08-01T15:59:33.966624+00:00 heroku[slack_server.1]: State changed from crashed to starting
2016-08-01T15:59:39.993589+00:00 heroku[slack_server.1]: Starting process with command `bundle exec rails runner bin/slack_server.rb`
2016-08-01T15:59:40.655054+00:00 heroku[slack_server.1]: State changed from starting to up
2016-08-01T15:59:45.275205+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Starting the New Relic agent in "production" environment.
2016-08-01T15:59:45.275246+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : To prevent agent startup add a NEW_RELIC_AGENT_ENABLED=false environment variable or modify the "production" section of your newrelic.yml.
2016-08-01T15:59:45.275247+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Reading configuration from config/newrelic.yml (/app)
2016-08-01T15:59:45.280046+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Environment: production
2016-08-01T15:59:45.280080+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : No known dispatcher detected.
2016-08-01T15:59:45.280110+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Application: VG Pizza
2016-08-01T15:59:45.318022+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Installing deferred Rack::Builder instrumentation
2016-08-01T15:59:45.318166+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Installing Rack::Builder middleware instrumentation
2016-08-01T15:59:45.319382+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Installing Sidekiq instrumentation
2016-08-01T15:59:45.319557+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Installing ActiveRecord 4+ instrumentation
2016-08-01T15:59:45.332822+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Installing ActiveJob instrumentation
2016-08-01T15:59:45.345692+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Installing Rails 3+ middleware instrumentation
2016-08-01T15:59:45.346276+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Installing Net instrumentation
2016-08-01T15:59:45.348721+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Installing Redis Instrumentation
2016-08-01T15:59:45.349027+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Installing Rails 4 view instrumentation
2016-08-01T15:59:45.349145+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Installing Rails 4 Controller instrumentation
2016-08-01T15:59:45.349283+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Installing Rails 4 Error instrumentation
2016-08-01T15:59:45.349354+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:45 +0000 slack_server.1 (3)] INFO : Finished instrumentation
2016-08-01T15:59:46.222921+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:46 +0000 slack_server.1 (3)] INFO : Installing Sinatra instrumentation
2016-08-01T15:59:46.310275+00:00 app[slack_server.1]:   User Load (7.2ms)  SELECT  "users".* FROM "users"  ORDER BY "users"."id" ASC LIMIT 1000
2016-08-01T15:59:46.334842+00:00 app[slack_server.1]: I, [2016-08-01T15:59:46.334748 #3]  INFO -- :  adding bot <VinnyBot key:xoxb-xxx>
2016-08-01T15:59:46.336049+00:00 app[slack_server.1]: I, [2016-08-01T15:59:46.336001 #3]  INFO -- :  adding bot <VinnyBot key:xoxb-xxx>
2016-08-01T15:59:46.337910+00:00 app[slack_server.1]: I, [2016-08-01T15:59:46.337870 #3]  INFO -- :  adding bot <VinnyBot key:xoxb-xxx>
2016-08-01T15:59:46.339088+00:00 app[slack_server.1]: I, [2016-08-01T15:59:46.339047 #3]  INFO -- :  adding bot <VinnyBot key:xoxb-xxx>
2016-08-01T15:59:46.339577+00:00 app[slack_server.1]: I, [2016-08-01T15:59:46.339538 #3]  INFO -- :  adding bot <VinnyBot key:xoxb-xxx>
2016-08-01T15:59:46.593135+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:46 +0000 slack_server.1 (3)] INFO : Reporting to: https://rpm.newrelic.com/accounts/xxx/applications/xxx
2016-08-01T15:59:48.116231+00:00 app[slack_server.1]: ** [NewRelic][2016-08-01 15:59:48 +0000 slack_server.1 (3)] INFO : Starting Agent shutdown
2016-08-01T15:59:48.251224+00:00 app[slack_server.1]: /app/vendor/bundle/ruby/2.3.0/gems/slack-bot-server-0.4.6/lib/slack_bot_server/bot.rb:181:in `rescue in start': account_inactive (SlackBotServer::Bot::ConnectionError)
2016-08-01T15:59:48.251227+00:00 app[slack_server.1]: 	from /app/vendor/bundle/ruby/2.3.0/gems/slack-bot-server-0.4.6/lib/slack_bot_server/bot.rb:152:in `start'
2016-08-01T15:59:48.251228+00:00 app[slack_server.1]: 	from /app/vendor/bundle/ruby/2.3.0/gems/slack-bot-server-0.4.6/lib/slack_bot_server/server.rb:82:in `block (2 levels) in start'
2016-08-01T15:59:48.251229+00:00 app[slack_server.1]: 	from /app/vendor/bundle/ruby/2.3.0/gems/slack-bot-server-0.4.6/lib/slack_bot_server/server.rb:82:in `each'
2016-08-01T15:59:48.251230+00:00 app[slack_server.1]: 	from /app/vendor/bundle/ruby/2.3.0/gems/slack-bot-server-0.4.6/lib/slack_bot_server/server.rb:82:in `block in start'
2016-08-01T15:59:48.251233+00:00 app[slack_server.1]: 	from /app/vendor/bundle/ruby/2.3.0/gems/eventmachine-1.2.0.1/lib/eventmachine.rb:194:in `run_machine'
2016-08-01T15:59:48.251234+00:00 app[slack_server.1]: 	from /app/vendor/bundle/ruby/2.3.0/gems/eventmachine-1.2.0.1/lib/eventmachine.rb:194:in `run'
2016-08-01T15:59:48.251234+00:00 app[slack_server.1]: 	from /app/vendor/bundle/ruby/2.3.0/gems/slack-bot-server-0.4.6/lib/slack_bot_server/server.rb:80:in `start'
2016-08-01T15:59:48.251236+00:00 app[slack_server.1]: 	from bin/slack_server.rb:26:in `<top (required)>'
2016-08-01T15:59:48.251236+00:00 app[slack_server.1]: 	from /app/vendor/bundle/ruby/2.3.0/gems/railties-4.2.5.1/lib/rails/commands/runner.rb:60:in `load'
2016-08-01T15:59:48.251237+00:00 app[slack_server.1]: 	from /app/vendor/bundle/ruby/2.3.0/gems/railties-4.2.5.1/lib/rails/commands/runner.rb:60:in `<top (required)>'
2016-08-01T15:59:48.251238+00:00 app[slack_server.1]: 	from /app/vendor/bundle/ruby/2.3.0/gems/railties-4.2.5.1/lib/rails/commands/commands_tasks.rb:123:in `require'
2016-08-01T15:59:48.251240+00:00 app[slack_server.1]: 	from /app/vendor/bundle/ruby/2.3.0/gems/railties-4.2.5.1/lib/rails/commands/commands_tasks.rb:123:in `require_command!'
2016-08-01T15:59:48.251241+00:00 app[slack_server.1]: 	from /app/vendor/bundle/ruby/2.3.0/gems/railties-4.2.5.1/lib/rails/commands/commands_tasks.rb:90:in `runner'
2016-08-01T15:59:48.251242+00:00 app[slack_server.1]: 	from /app/vendor/bundle/ruby/2.3.0/gems/railties-4.2.5.1/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
2016-08-01T15:59:48.251243+00:00 app[slack_server.1]: 	from /app/vendor/bundle/ruby/2.3.0/gems/railties-4.2.5.1/lib/rails/commands.rb:17:in `<top (required)>'
2016-08-01T15:59:48.251245+00:00 app[slack_server.1]: 	from bin/rails:4:in `require'
2016-08-01T15:59:48.251245+00:00 app[slack_server.1]: 	from bin/rails:4:in `<main>'
2016-08-01T15:59:48.398640+00:00 heroku[slack_server.1]: Process exited with status 1
2016-08-01T15:59:48.418994+00:00 heroku[slack_server.1]: State changed from up to crashed
```

**Comment by lazyatom** [comment]
Apologies for the delay in responding here; I believe this was already fixed in master, but I've now released version 0.4.7 to rubygems.org, which includes the fix too. Please do let me know if this doesn't resolve your issue!

**Comment by gabeodess** [comment]
@lazyatom yeah, once I started digging through the code I saw that on the master branch.  Thanks for the release!
